site/guides/integration-testing.adoc

```diff
@@ -0,0 +1,970 @@
+
+[source,xml]
+----
+  <dependencies>
+    <dependency>
+      <groupId>org.jboss.arquillian.junit</groupId>
+      <artifactId>arquillian-junit-standalone</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.arquillian.cube</groupId>
+      <artifactId>arquillian-cube-openshift</artifactId>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>io.undertow</groupId>
+          <artifactId>undertow-core</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency
+   <dependencies>
+----
+
+The dependencies above provide add the minimum number of arquillian dependencies needed to interact with the application when it's deployed on OpenShift.
+
+==== Maven failsafe plugin configuration
+
+Making sure that the integration tests only run in a proper environment is very important. We don't want developers inadvertently launching the integration tests
+by accident only to have them fail because there is no OpenShift cluster available and thus creating the false impression that the build is broken.
+
+For that reason we introduce a Maven profile named `openshift-it` which will contain the failsafe plugin configuration needed to run the tests. +
+This means that the integration tests will only run when the profile is activated using `-Popenshift-it` as an argument to Maven's command-line invocation.
+
+The profiles section of `pom.xml` will now be:
+
+[source, xml]
+----
+  <profiles>
+    <profile>
+      <id>openshift-it</id>  #<1>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-failsafe-plugin</artifactId> #<2>
+            <version>2.20</version>
+            <configuration>
+              <systemPropertyVariables>
+                <app.name>${project.artifactId}</app.name> #<3>
+              </systemPropertyVariables>
+            </configuration>
+            <executions>
+              <execution>
+                <goals>
+                  <goal>integration-test</goal>
+                  <goal>verify</goal>           #<4>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  <profiles>
+----
+
+<1> The `openshift-it` profile is declared
+
+<2> The failsafe plugin is made part of the project's build when the `openshift-it` profile is enabled
+
+<3> We add a system property named `app.name` that will be available to any piece of code running as part of the failsafe plugin's lifecycle. +
+The value of this property is set to the project's `artifactId` as defined in its POM file and will be used later in the test code to allow Arquillian to access the application. +
+For that reason it is crucial that the value of the `app.name` property matches exactly the name of the `Route` used for the application under test.
```

Comment:
Indeed, this is a very good comment. I will initially explain why the route uses that name and then we can see if we want to use explicit configuration instead of the defaults